default/python/AI/test/abstract_test_bases.py

```diff
@@ -0,0 +1,61 @@
+import unittest
+
+
+class PropertyTester(unittest.TestCase):
```

Comment:
Please add a docstring. 

Mention that object is mutated.
In my mock generator, I have encountered a problem that not all classes can be instantiated by python.  I used function to get real objecst from the game.   In that case mutation might be a problem and require ever cleanup or game exit after tests.